File: skellig-test-step-processing-jdbc/src/main/java/org/skellig/teststep/processor/jdbc/BaseJdbcRequestExecutor.java
package org.skellig.teststep.processor.jdbc;

import org.skellig.teststep.processor.db.BaseDatabaseRequestExecutor;

import java.sql.Date;
import java.sql.Timestamp;
import java.time.Instant;
import java.time.LocalDate;
import java.time.LocalDateTime;

abstract class BaseJdbcRequestExecutor extends BaseDatabaseRequestExecutor {

    @Override
    protected Object getParameterValue(Object item) {
        item = super.getParameterValue(item);

        if (item instanceof LocalDateTime) {
            return fromLocalDateTimeToTimestamp((LocalDateTime) item);
        } else if (item instanceof LocalDate) {
            return fromLocalDateToSqlDate((LocalDate) item);
        } else if (item instanceof Instant) {
            return fromInstantToTimestamp((Instant) item);
        } else {
            return item;
        }
    }

    private Timestamp fromLocalDateTimeToTimestamp(LocalDateTime localDateTime) {
        return Timestamp.valueOf(localDateTime);
    }

    private Date fromLocalDateToSqlDate(LocalDate localDate) {
        return Date.valueOf(localDate);
    }

    private Timestamp fromInstantToTimestamp(Instant instant) {
        return fromLocalDateTimeToTimestamp(LocalDateTime.from(instant));
    }
}

File: skellig-test-step-processing-jdbc/src/main/kotlin/org/skellig/teststep/processor/jdbc/BaseJdbcRequestExecutor.kt
package org.skellig.teststep.processor.jdbc

import org.skellig.teststep.processor.db.BaseDatabaseRequestExecutor
import java.sql.Date
import java.sql.Timestamp
import java.time.Instant
import java.time.LocalDate
import java.time.LocalDateTime

internal abstract class BaseJdbcRequestExecutor : BaseDatabaseRequestExecutor() {

    override fun getParameterValue(item: Any?): Any? {
        var newItem = item
        newItem = super.getParameterValue(newItem)
        return when (item) {
            is LocalDateTime -> fromLocalDateTimeToTimestamp(newItem!! as LocalDateTime)
            is LocalDate -> fromLocalDateToSqlDate(newItem!! as LocalDate)
            is Instant -> fromInstantToTimestamp(newItem!! as Instant)
            else -> newItem
        }
    }

    private fun fromLocalDateTimeToTimestamp(localDateTime: LocalDateTime): Timestamp {
        return Timestamp.valueOf(localDateTime)
    }

    private fun fromLocalDateToSqlDate(localDate: LocalDate): Date {
        return Date.valueOf(localDate)
    }

    private fun fromInstantToTimestamp(instant: Instant): Timestamp {
        return fromLocalDateTimeToTimestamp(LocalDateTime.from(instant))
    }
}